test/gtest/uct/test_md.cc

```diff
@@ -242,6 +242,32 @@ UCS_TEST_P(test_md, mem_type_detect_mds) {
     }
 }
 
+UCS_TEST_P(test_md, mem_query) {
+
+    uct_md_attr_t md_attr;
+    ucs_status_t status;
+    uct_md_mem_attr_t mem_attr;
+    uint8_t mem_type_id;
+    void *address;
+
+    status = uct_md_query(md(), &md_attr);
+    ASSERT_UCS_OK(status);
+
+    if (!md_attr.cap.detect_mem_types) {
+        UCS_TEST_SKIP_R("MD can't detect any memory types");
+    }
+
+    ucs_for_each_bit(mem_type_id, md_attr.cap.detect_mem_types) {
+        alloc_memory(&address, UCS_KBYTE, NULL,
+                     static_cast<ucs_memory_type_t>(mem_type_id));
+        status = uct_md_mem_query(md(), address, UCS_KBYTE, &mem_attr);
+        ASSERT_UCS_OK(status);
+        ASSERT_NE(mem_attr.field_mask, 0);
+        EXPECT_TRUE(mem_attr.field_mask & UCT_MD_MEM_ATTR_FIELD_MEM_TYPE);
```

Comment:
I think we should only expect that MEM_TYPE bit is set.
could you remove `ASSERT_NE(...)` as it duplicated, please? `EXPECT_TRUE()` will catch error in either way